src/services/services.ts

```diff
@@ -7339,11 +7333,17 @@ namespace ts {
                     if (declarations && declarations.length > 0) {
                         // Disallow rename for elements that are defined in the standard TypeScript library.
                         const defaultLibFileName = host.getDefaultLibFileName(host.getCompilationSettings());
+                        const canonicalDefaultLibName = getCanonicalFileName(ts.normalizePath(defaultLibFileName));
                         if (defaultLibFileName) {
                             for (const current of declarations) {
                                 const sourceFile = current.getSourceFile();
+                                // TODO (drosen): When is there no source file?

```

Comment:
To answer: When you've fooled the language service by injecting a declaration node which is not parented to anything, yet is associated with a symbol returned by `getSymbolAtLocation` - which could conceivably happen if you wrapped the language service and injected declarations for templated components, but have not rewritten the `getSourceFile` implementation to do something other than walk up the tree? TBH, it probably never happens under normal conditions and is likely exceptional - every node should be nested within a source file in a well-behaved program. Maybe it would be more correct to toss in a `Debug.fail` here, despite that not being the old behavior?